framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -2767,4 +2770,36 @@ public Boolean getEnsuresQualifierIfResult(
         AnnotationUtils.getElementValueArray(contractAnnotation, elementName, String.class, true);
     return result;
   }
+
+  /**
+   * Returns the type factory for the given annotation processor, if it is type-checker.
+   *
+   * @param processorName the fully-qualified class name of an annotation processor
+   * @return the type factory for the given annotation processor, or null if it's not a checker
+   */
+  public GenericAnnotatedTypeFactory<?, ?, ?, ?> getTypeFactory(@BinaryName String processorName) {
```

Comment:
I think I would rename the method to `createTypeFactoryForProcessor` or something similar.  This method isn't getting the type factory, it's creating it.